workers/deployment_manager/DeploymentManager/DeploymentService/DeploymentModifier.cs

```diff
@@ -0,0 +1,210 @@
+            };
+
+            deploymentServiceClient.UpdateDeployment(request);
+        }
+
+        public static void RemoveDeploymentTag(string deploymentId, string projectName, string tag)
+        {
+            var deployment = GetDeployment(deploymentId, projectName);
+
+            if (deployment.Tag.Contains(tag))
+            {
+                deployment.Tag.Remove(tag);
+            }
+
+            var request = new UpdateDeploymentRequest
+            {
+                Deployment = deployment
+            };
+
+            deploymentServiceClient.UpdateDeployment(request);
+        }
+
+        public static void UpdateDeploymentTag(string deploymentId, string projectName, string tagPrefix, string value)
+        {
+            var deployment = GetDeployment(deploymentId, projectName);
+            var tags = deployment.Tag.ToList();
+            foreach (var tag in tags)
+            {
+                if (tag.Contains(tagPrefix))
+                {
+                    RemoveDeploymentTag(deploymentId, projectName, tag);
+                }
+            }
+
+            AddDeploymentTag(deploymentId, projectName, $"{tagPrefix}_{value}");
+        }
+
+        public static Google.LongRunning.Operation<Deployment, CreateDeploymentMetadata> CreateDeployment(DeploymentTemplate template)
+        {
+                var request = new CreateDeploymentRequest
+                {
+                    Deployment = new Deployment
+                    {
+                        AssemblyId = template.AssemblyId,
+                        LaunchConfig = template.LaunchConfig,
+                        Name = template.DeploymentName,
+                        ProjectName = template.ProjectName,
+                        StartingSnapshotId = template.SnapshotId,
+                    }
+                };
+
+                return deploymentServiceClient.CreateDeployment(request);
+        }
+
+        public static void StopDeployment(Deployment deployment)
+        {
+                var request = new StopDeploymentRequest
+                {
+                    Id = deployment.Id,
+                    ProjectName = deployment.ProjectName,
+                };
+
+                deploymentServiceClient.StopDeployment(request);
+        }
+
+        public static List<Deployment> ListDeployments(string projectName)
+        {
+            var request = new ListDeploymentsRequest
+            {
+                ProjectName = projectName,
+            };
+
+            var response = deploymentServiceClient.ListDeployments(request);
+            return response.ToList();
+        }
+
+        public static int GetMaxWorkerCapacity(string deploymentId, string projectName, string workerType)
+        {
+            var deployment = GetDeployment(deploymentId, projectName);
+            foreach (var workerCapacity in deployment.WorkerConnectionCapacities)
+            {
+                if (workerCapacity.WorkerType.Equals(workerType))
```

Comment:
Why not `==`?

Reply:
still a habit from using java where `==` for strings is wrong. :P really need to stop doing that